tests/unit/processing/test_dataset_metadata_update.py

```diff
@@ -55,212 +54,14 @@ def mock_download(source_uri, local_path):
-                DatasetStatusUpdate(status_key=DatasetStatusKey.RDS, status=rds_status),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.CXG, status=DatasetConversionStatus.CONVERTED),
-            ]
-        )
-
-        dataset_version_id = DatasetVersionId(dataset_version.dataset_version_id)
-        assert self.updater.has_valid_artifact_statuses(dataset_version_id) is False
-
-    @parameterized.expand([DatasetConversionStatus.CONVERTING, DatasetConversionStatus.FAILED])
-    def test_has_valid_artifact_statuses__invalid_h5ad_status(self, h5ad_status):
-        dataset_version = self.generate_dataset(
-            statuses=[
-                DatasetStatusUpdate(status_key=DatasetStatusKey.PROCESSING, status=DatasetProcessingStatus.PENDING),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.H5AD, status=h5ad_status),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.RDS, status=DatasetConversionStatus.CONVERTED),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.CXG, status=DatasetConversionStatus.CONVERTED),
-            ]
-        )
-
-        dataset_version_id = DatasetVersionId(dataset_version.dataset_version_id)
-        assert self.updater.has_valid_artifact_statuses(dataset_version_id) is False
-
-    @parameterized.expand([DatasetConversionStatus.CONVERTING, DatasetConversionStatus.FAILED])
-    def test_has_valid_artifact_statuses__invalid_cxg_status(self, cxg_status):
-        dataset_version = self.generate_dataset(
-            statuses=[
-                DatasetStatusUpdate(status_key=DatasetStatusKey.PROCESSING, status=DatasetProcessingStatus.PENDING),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.H5AD, status=DatasetConversionStatus.CONVERTED),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.RDS, status=DatasetConversionStatus.CONVERTED),
-                DatasetStatusUpdate(status_key=DatasetStatusKey.CXG, status=cxg_status),
-            ]
-        )
-
-        dataset_version_id = DatasetVersionId(dataset_version.dataset_version_id)
-        assert self.updater.has_valid_artifact_statuses(dataset_version_id) is False
-
-    @patch("backend.common.utils.dl_sources.uri.downloader")
-    @patch("scanpy.read_h5ad")
-    @patch("backend.layers.processing.dataset_metadata_update.DatasetMetadataUpdater.update_h5ad")
-    @patch("backend.layers.processing.dataset_metadata_update.DatasetMetadataUpdater.update_rds")
-    @patch("backend.layers.processing.dataset_metadata_update.DatasetMetadataUpdater.update_cxg")
+    @patch("backend.layers.processing.dataset_metadata_update.S3Provider", Mock(side_effect=MockS3Provider))
+    @patch("backend.layers.processing.dataset_metadata_update.DatabaseProvider", Mock(side_effect=DatabaseProviderMock))
+    @patch("backend.layers.processing.dataset_metadata_update.DatasetMetadataUpdaterWorker.update_h5ad")
```

Comment:
Why not mock the class `DatasetMetadataUpdaterWorker` and check if the functions are called? Then you don't you don't need to mock each one individually